### PR TITLE
Mobile rewards

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: app/src/component/deviceIdRewardSubcard/view.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  render() {
+    return (
+      <View style={rewardStyle.subcard}>
+        <Text style={rewardStyle.subtitle}>Pending action: Device ID</Text>
+        <Text style={[rewardStyle.bottomMarginMedium, rewardStyle.subcardText]}>
+          The app requires the phone state permission in order to identify your device for reward eligibility.
+        </Text>
+        <Button style={rewardStyle.actionButton}
```
Comment (skhameneh) [inline app/src/component/deviceIdRewardSubcard/view.js]: Adjust formatting, I'd suggest:
```
        <Button
          style={rewardStyle.actionButton}
          text={"Allow Access"}
          onPress={this.onAllowAccessPressed}
        />
```

File: app/src/component/emailRewardSubcard/view.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    AsyncStorage.getItem(Constants.KEY_FIRST_RUN_EMAIL).then(email => {
+      if (email && email.trim().length > 0) {
+        this.setState({ email });
+      } else {
+        this.setState({ email: emailToVerify });
+      }
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (skhameneh) [inline app/src/component/emailRewardSubcard/view.js]: No more legacy methods!
https://github.com/lbryio/lbry-android/pull/232/files#r210515192

Comment (skhameneh) [reply app/src/component/emailRewardSubcard/view.js]: See the big yellow box:
https://reactjs.org/docs/react-component.html#updating

File: app/src/component/rewardCard/view.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    reward_type: string,
+  },
+};
+
+class RewardCard extends React.PureComponent<Props> {
+  state = {
+    claimStarted: false
+  };
+
+  componentWillReceiveProps(nextProps) {
```
Comment (skhameneh) [inline app/src/component/rewardCard/view.js]: https://reactjs.org/docs/react-component.html#updating

File: app/src/page/rewards/view.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        this.setState({ canAcquireDeviceId });
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    DeviceEventEmitter.removeListener('onPhoneStatePermissionGranted', this.phoneStatePermissionGranted);
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (skhameneh) [inline app/src/page/rewards/view.js]: https://reactjs.org/docs/react-component.html#updating

File: app/src/component/emailRewardSubcard/view.js (hunk 2 above)
Comment (akinwale) [reply app/src/component/emailRewardSubcard/view.js]: Heh! The reason I haven't moved on from this yet is we're still using an older React version, 16.2.0. I've read the documentation and it looks like the warnings started appearing from v16.3.0. I haven't had the chance to explore upgrading to newer versions of react yet (there were some issues the last time I tried), but I would love to discard all legacy methods when the time comes.

**Pull request merged**